pymedphys/_dicom/anonymisation_strategy.py

```diff
@@ -56,97 +56,30 @@ def _get_vr_anonymous_hardcode_replacement_value(value_representation, current_v
-
-
-def _get_OB_or_OW_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("OB or OW", current_value)
-
-
-def _get_OW_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("OW", current_value)
-
-
-def _get_PN_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("PN", current_value)
-
-
-def _get_SH_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("SH", current_value)
-
-
-def _get_SQ_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("SQ", current_value)
-
-
-def _get_ST_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("ST", current_value)
-
-
-def _get_TM_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("TM", current_value)
-
-
-def _get_UI_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("UI", current_value)
-
-
-def _get_US_anonymous_hardcode_replacement_value(current_value):
-    return _get_vr_anonymous_hardcode_replacement_value("US", current_value)
-
-
-ANONYMISATION_HARDCODE_DISPATCH = dict(
-    {
-        "AE": _get_AE_anonymous_hardcode_replacement_value,
-        "AS": _get_AS_anonymous_hardcode_replacement_value,
-        "CS": _get_CS_anonymous_hardcode_replacement_value,
-        "DA": _get_DA_anonymous_hardcode_replacement_value,
-        "DS": _get_DS_anonymous_hardcode_replacement_value,
-        "DT": _get_DT_anonymous_hardcode_replacement_value,
-        "LO": _get_LO_anonymous_hardcode_replacement_value,
-        "LT": _get_LT_anonymous_hardcode_replacement_value,
-        "OB": _get_OB_anonymous_hardcode_replacement_value,
-        "OB or OW": _get_OB_or_OW_anonymous_hardcode_replacement_value,
-        "OW": _get_OW_anonymous_hardcode_replacement_value,
-        "PN": _get_PN_anonymous_hardcode_replacement_value,
-        "SH": _get_SH_anonymous_hardcode_replacement_value,
-        "SQ": _get_SQ_anonymous_hardcode_replacement_value,
-        "ST": _get_ST_anonymous_hardcode_replacement_value,
-        "TM": _get_TM_anonymous_hardcode_replacement_value,
-        "UI": _get_UI_anonymous_hardcode_replacement_value,
-        "US": _get_US_anonymous_hardcode_replacement_value,
-    }
-)
+DISPATCH_KEYS_TO_OVERRIDE = [
+    "AE",
+    "AS",
+    "CS",
+    "DA",
+    "DS",
+    "DT",
+    "LO",
+    "LT",
+    "OB",
+    "OB or OW",
+    "OW",
+    "PN",
+    "SH",
+    "SQ",
+    "ST",
+    "TM",
+    "UI",
+    "US",
+]
+
+ANONYMISATION_HARDCODE_DISPATCH = {
+    key: functools.partial(
+        _get_vr_anonymous_hardcode_replacement_value, value_representation=key
+    )
+    for key in DISPATCH_KEYS_TO_OVERRIDE
+}
```

Comment:
After your comments regarding having it not really worth detailing `DISPATCH_KEYS_TO_OVERRIDE` given that's just the VR, would it be better to do the following:

```python
def get_default_hardcode_dispatch():
    keys = get_vr_anonymous_replacement_value_dict().keys()

	ANONYMISATION_HARDCODE_DISPATCH = {
	    key: functools.partial(
	        _get_vr_anonymous_hardcode_replacement_value, value_representation=key
	    )
	    for key in DISPATCH_KEYS_TO_OVERRIDE
	}

	return ANONYMISATION_HARDCODE_DISPATCH
```

Reply:
I've done half of this, without using keys as a temporary variable.
If we wrap the get_default_hardcode_dispatch with a lru_cache, then it avoids the dispatch table being built until it is first used.  If you want that done, I'm OK with it.  But if there is a problem, I'd like it to fail fast, and it will fail fast if it's a module level variable declaration (the first time someone imports the module... it will fail then if it's going to fail).

Reply:
And now I've done the other half, include lru_cache.
I don't understand why it failed to find pydicom, but that was enough motivation for me to complete the approach you suggested and hope that runs the gauntlet

Reply:
and used the temporary variable to avoid pylint whinging.

Reply:
> I don't understand why it failed to find pydicom

Because by design, the PyMedPhys CLI and library needs to be able to start up and run even if none of its dependencies are installed. This is to support stripped down installations in weird and wonderful places such as [MicroPython](https://micropython.org/) and [PyOdide](https://hacks.mozilla.org/2019/04/pyodide-bringing-the-scientific-python-stack-to-the-browser/)

Reply:
To give some more info here, before any dependencies are installed the following two commands a run to make sure that they at least run without error:

```bash
pymedphys --help
python -c "import pymedphys"
```

For more details, see:

https://github.com/pymedphys/pymedphys/blob/1086964c25f569f2171074f70f58076520c4f5b9/.github/workflows/on-push.yml#L119-L141